src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        /// <summary>
+        /// Convenience constructor for the scalar case.
+        /// Min and Max are set to the single value ordinal.
+        /// When a given column spans only a single column in the
+        /// dataset.
```

Comment:
This is not a sentence. It almost works if you were to append it to the *first* sentence, followed by a comma.

Reply:
I concur, as a non-native speaker, I'd go to read the code (and be slightly annoyed). :)

Reply:
Still not a sentence. Perhaps reading this summary comment aloud to yourself would help?

---
In reply to: [189083569](https://github.com/dotnet/machinelearning/pull/142#discussion_r189083569) [](ancestors = 189083569)

Reply:
Oh OK... I see what went wrong here.

Append this third "sentence" to the first sentence followed by a comma.

By this I mean, I am **not** proposing the following transformation.

"I like to run. When the sun shines." => "I like to run. When the sun shines," because that's still not a sentence and its punctuation makes no sense.

I am proposing the following transformation.

"I like to run, when the sun shines."

---
In reply to: [189340150](https://github.com/dotnet/machinelearning/pull/142#discussion_r189340150) [](ancestors = 189340150,189083569)